src/components/input_field.tsx

```diff
@@ -42,152 +50,42 @@ export interface InputFieldProps {
-        }
-    }
-
-    _showValidationError() {
-        const { validationError, submitted } = this.props;
-        const { touched } = this.state;
-        return validationError && (touched || submitted)
-    }
-
-    private _renderLabel(): JSX.Element {
-        const { id, label, labelLayoutWidth, labelWidthBreakpoint, type, preLabelElement, infoText, infoLinkURL } = this.props;
-
-        if (label || preLabelElement) {
-            return <InputFieldLabel
-                type={type}
-                id={id}
-                label={label}
-                labelLayoutWidth={labelLayoutWidth}
-                labelWidthBreakpoint={labelWidthBreakpoint}
-                preLabelElement={preLabelElement}
-                infoText={infoText}
-                infoLinkURL={infoLinkURL}
-            />
-        }
-        return null;
-    }
-
-    private _renderInputElements() {
-        const { id, type, centerInputField, value, min, max, placeholder,
-            selectOptions, imageSize, staticField, imageShape, tagSuggestions, textareaAutoGrow,
-            focusOnLoad, required, labelLayoutWidth, labelWidthBreakpoint, children } = this.props;
-
-        return <InputFieldInput
-            type={type}
-            id={id}
-            centerInputField={centerInputField}
-            onChangeHandler={this._onChangeHandler}
-            value={value}
-            min={min}
-            max={max}
-            placeholder={placeholder}
-            focusOnLoad={focusOnLoad}
-            selectOptions={selectOptions}
-            imageSize={imageSize}
-            imageShape={imageShape}
-            staticField={staticField}
-            tagSuggestions={tagSuggestions}
-            required={required}
-            showValidationError={this._showValidationError()}
-            labelLayoutWidth={labelLayoutWidth}
-            labelWidthBreakpoint={labelWidthBreakpoint}
-            textareaAutoGrow={textareaAutoGrow}
-            children={children}
-        />
-    }
-
-    private _renderValidation(): JSX.Element {
-        const { validationError, labelLayoutWidth, labelWidthBreakpoint, required } = this.props;
+    private readonly input_fields = {
+        number: NumberField,
+        textarea: TextareaField,
+        select: SelectField,
+        checkbox: CheckboxField,
+        image: ImageUploadField,
+        date: DateField,
+        file: FileUploadField,
+        children: ChildrenField,
+        tags: TagsField,
+    };
 
-        if (this._showValidationError() && validationError) {
-            return <InputFieldValidation
-                validationError={validationError}
-                labelLayoutWidth={labelLayoutWidth}
-                labelWidthBreakpoint={labelWidthBreakpoint}
-                required={required}
-            />
+    render() {
+        const {type, value, children, ...wrapperProps} = this.props;
+        if (type === 'text' || type === 'password' || type === 'email' || type === 'url') {
```

Comment:
I wouldn't be against also just putting these types in the map above.

Reply:
We'd have to pass type to each input field but that's fine